src/c++/perf_analyzer/test_utils.h

```diff
@@ -26,13 +26,55 @@
 
 #pragma once
 
+#include <atomic>
 #include <cmath>
 #include <cstdint>
 #include <numeric>
+#include <thread>
 #include <vector>
 
 namespace triton { namespace perfanalyzer {
 
+// FIXME
+class WatchDog {
+ public:
+  WatchDog(unsigned int max_time_ms) { start(max_time_ms); }
+
+  void stop()
+  {
+    running_ = false;
+    thread_.join();
+  }
+
+ private:
+  uint interval{100};
+  uint max_time_ms_;
+  std::atomic<unsigned int> timer_;
+  std::atomic<bool> running_;
+  std::thread thread_;
+
+  void start(unsigned int max_time_ms)
+  {
+    max_time_ms_ = max_time_ms;
```

Comment:
Should there be some protection here for preventing `start()` from being called a second time before a paired call to `stop()`?

Reply:
start is a private function. If we decide to expand the watchdog to be reused with a public start function, we can explore it then